### PR TITLE
Stargazers are now station-restricted, and cogged APCs are now quiet

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
balance: Stargazers can no longer be built off-station, even in new areas.
balance: Integration cogs no longer emit sounds and steam visuals when active.
/:cl:

I haven't been playing or watching in person (other stuff has me occupied) lately, even though I'd like to, but I've been keeping a close eye on people's feedback for the mode, and the main problem that I've seen is the cheese strategy of having a giant stargazer box built in deep space far beyond where anyone could reach you.

This fixes that; stargazers now have to be built in an initial station area. This *does* include solars, but as such they can be scouted out and toolboxed by anyone with an EVA suit, as they were intended to be.

Integration cogs also had a problem: they were meant to be stealthy, but the extended sound range and frequent tells made them too easy to distinguish and thus defeated the purpose. They no longer have any hard tells, meaning that you have to at least check the APC's interface to determine if it's cogged.

### CODE REVIEW

File: code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm [hunk 1]
```diff
@@ -231,6 +231,13 @@
 	quickbind = TRUE
 	quickbind_desc = "Creates a stargazer, which generates power when near starlight."
 
+/datum/clockwork_scripture/create_object/stargazer/check_special_requirements()
+	var/area/A = get_area(invoker)
+	if(A.outdoors || A.map_name == "Space")
```
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm]: ew, don't check strings like this, it's bad and might not catch stuff properly.

Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm]: There doesn't seem to be a better way; if there is, it's a proc that I'm unaware of. The only thing I can think of is adding a new helper proc that does the same thing like `initial_space_area(turf)` or something that checks if the area is in space or was initially space.

Comment (Robustin) [reply code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm]: Xhuis I told you about the blob proc, I used it for the narsie rune too before I limited it to a randomized set of rooms.

Comment (Xhuis) [reply code/game/gamemodes/clock_cult/clock_scriptures/scripture_drivers.dm]: I did actually search for that, but I couldn't locate it in the code. If you know the syntax for it I'd be happy to include it.

**Comment by ChangelingRain** [comment]
Also I'm pretty sure this will just make the issues with the power system super evident, since stargazers are about as unsubtle of an object as you can make.
So, like, have fun trying to balance power amounts.

**Comment by Xhuis** [comment]
I've spoken about power  already in [this thread.](https://tgstation13.org/phpBB/viewtopic.php?f=10&t=13167) This won't amplify an illusionary problem that's perceived to exist; just further bring it to the intended function.

**Pull request merged**